### PR TITLE
Refactor manipulation routines from `core.pyx`

### PR DESCRIPTION
**Pull request by niboshi**
An attempt to slim down `core.pyx`.

### CODE REVIEW

**Comment by okuta** [comment]
I want to merge this PR. Please send message when you finished.

**Comment by niboshi** [comment]
@okuta 
I'm considering dividing `_ndarray_impl` further into routines, like `_routines/statistics.pyx` or `_routines/indexing.pyx`. Do you think it's fine?

**Comment by okuta** [comment]
`core/_routines_statistics.pyx` is fine as a temporary because there are not many routines.

**Comment by niboshi** [comment]
I've redone the fix to `reshape` and `ravel`.

Btw, I think it's better to proceed the refactoring one-by-one because it would be time consuming to refactor all the methods and extremely difficult to follow master changes.

If agreed, I'll try to refactor other `manipulation` methods.

File: tests/cupy_tests/core_tests/test_core.py [hunk 1]
```diff
@@ -5,23 +5,6 @@
 from cupy import testing
 
 
-class TestGetStridesForNocopyReshape(unittest.TestCase):
```
Comment (niboshi) [inline tests/cupy_tests/core_tests/test_core.py]: Btw, this test has been removed because `_get_strides_for_nocopy_reshape` is no longer exposed now.

Comment (okuta) [reply tests/cupy_tests/core_tests/test_core.py]: OK.

File: cupy/core/__init__.py [hunk 2]
```diff
@@ -38,20 +42,16 @@
 from cupy.core.core import less  # NOQA
 from cupy.core.core import less_equal  # NOQA
 from cupy.core.core import matmul  # NOQA
-from cupy.core.core import moveaxis  # NOQA
 from cupy.core.core import multiply  # NOQA
 from cupy.core.core import nanmax  # NOQA
 from cupy.core.core import nanmin  # NOQA
 from cupy.core.core import ndarray  # NOQA
 from cupy.core.core import negative  # NOQA
-from cupy.core.core import normalize_axis_tuple  # NOQA
```
Comment (niboshi) [inline cupy/core/__init__.py]: Also made `normalize_axis_tuple` private because it's not used from anywhere.

**Comment by niboshi** [comment]
For `manipulation` routines I've finished dissecting.
Please review.

**Comment by okuta** [comment]
jenkins, test this please.

**Comment by niboshi** [comment]
Fixed to follow master changes.

**Comment by niboshi** [comment]
Rebased once again.

**Comment by okuta** [comment]
thanks

jenkins, test this please.

**Comment by okuta** [comment]
LGTM!

**Pull request merged**